gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -2227,71 +2271,30 @@ void Movable::FixPosition()
 	Pos.y=Pos.y*12+6;
 }
 
-void Movable::WalkTo(const Point &Des, int distance)
-{
-	Point from;
-
-	// maybe caller should be responsible for this
-	if ((Des.x/16 == Pos.x/16) && (Des.y/12 == Pos.y/12)) {
-		ClearPath();
-		return;
-	}
+void Movable::WalkTo(const Point &Des, int distance) {
+	Log(DEBUG, "WalkToWIP", "curTime = %d, prevTicks = %d", Ticks, this->prevTicks);
+	// The rate limiting stuff is probably not optimal, since it requires lots of pathfinder calls
+	// An incremental pathfinding algorithm would be preferable, but LOS checks are slow
+	// and not pre-cachable, while incremental pathfinding requires quite a lot of checks
+	if (this->Ticks > this->prevTicks) { // Rate limiting
+		this->prevTicks = Ticks;
 
-	// the prev_step stuff is a naive attempt to allow re-pathing while moving
-	PathNode *prev_step = NULL;
-	unsigned char old_stance = StanceID;
-	if (step && step->Next) {
-		// don't interrupt in the middle of a step; path from the next one
-		prev_step = new PathNode(*step);
-		from.x = ( step->Next->x * 16 ) + 8;
-		from.y = ( step->Next->y * 12 ) + 6;
-	}
+		const int XEPS = 2;
+		const int YEPS = 1;
 
-	ClearPath();
-	if (!prev_step) {
-		FixPosition();
-		from = Pos;
-	}
-	area->ClearSearchMapFor(this);
-	if (distance) {
-		path = area->FindPathNear( from, Des, size, distance );
-	} else {
-		path = area->FindPath( from, Des, size, distance );
-	}
-	//ClearPath sets destination, so Destination must be set after it
-	//also we should set Destination only if there is a walkable path
-	if (path) {
-		Destination = Des;
-
-		if (prev_step) {
-			// we want to smoothly continue, please
-			// this all needs more thought! but it seems to work okay
-			StanceID = old_stance;
-
-			if (path->Next) {
-				// this is a terrible hack to make up for the
-				// pathfinder orienting the first node wrong
-				// should be fixed in pathfinder and not here!
-				Point next, follow;
-				next.x = path->x; next.y = path->y;
-				follow.x = path->Next->x;
-				follow.y = path->Next->y;
-				path->orient = GetOrient(follow, next);
-			}
+		// Callers should definitely be responsible for this
+		// The sooner this is checked, the better
```

Comment:
unless you mean with an intermediary function, I don't see how this is the caller's duty. Now the interface is very clear — walk close enough to this point.